build/android/bytecode/java/org/brave/bytecode/BraveBaseCustomTabActivityClassAdapter.java

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.brave.bytecode;
+
+import org.objectweb.asm.ClassVisitor;
+
+public class BraveBaseCustomTabActivityClassAdapter extends BraveClassVisitor {
+    static String sBaseCustomTabActivityClassName =
+            "org/chromium/chrome/browser/customtabs/BaseCustomTabActivity";
```

Comment:
why do we need to modify things in a base class for custom tab activity? Isn't this designed to be subclasses?

Reply:
I'm not sure what we're trying to do here, but we are definitely misusing this class and this is not how we should be using asm.

Reply:
so for instance if we need a brave version of `BaseCustomTabRootUiCoordinator` we can override `BaseCustomTabActivity` to use that instead, but we shouldn't be changing the this class to allow any arbitrary `mBaseCustomTabRootUiCoordinator` to be substituted here because that violates the contract this class provides.

Reply:
Are we really doing all of this just for this? https://github.com/brave/brave-core/pull/26945/files#r1950048063

Reply:
yes. we needed to override `mBaseCustomTabRootUiCoordinator` only for `RewardsPageActivity`. do you have any other suggestions ?

Reply:
We can probably set visibility GONE for `control_container` in `RewardsPageActivity`, just like we do for `toolbar_container`, if all we need is its height to be 0.

Reply:
@samartnik i explored your suggestion but it wouldn't work as with `control_container ` with visibility as `GONE` would hide the UI but we have a top margin set in BaseCustomActivity and for that we need to change the dimens.